tests/e2e/specs/miscellaneous/CustomOpenVSXRegistry.spec.ts

```diff
@@ -0,0 +1,222 @@
+/** *******************************************************************
+ * copyright (c) 2023 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+import { FACTORY_TEST_CONSTANTS } from '../../constants/FACTORY_TEST_CONSTANTS';
+import { BASE_TEST_CONSTANTS } from '../../constants/BASE_TEST_CONSTANTS';
+import { BrowserTabsUtil } from '../../utils/BrowserTabsUtil';
+import { e2eContainer } from '../../configs/inversify.config';
+import { CLASSES, TYPES } from '../../configs/inversify.types';
+import { WorkspaceHandlingTests } from '../../tests-library/WorkspaceHandlingTests';
+import { ProjectAndFileTests } from '../../tests-library/ProjectAndFileTests';
+import { LoginTests } from '../../tests-library/LoginTests';
+import { ActivityBar, SideBarView, ViewItem, ViewSection } from 'monaco-page-objects';
+import { registerRunningWorkspace } from '../MochaHooks';
+import { Logger } from '../../utils/Logger';
+import { expect } from 'chai';
+import { ShellString } from 'shelljs';
+import { KubernetesCommandLineToolsExecutor } from '../../utils/KubernetesCommandLineToolsExecutor';
+import { ITestWorkspaceUtil } from '../../utils/workspace/ITestWorkspaceUtil';
+import { TIMEOUT_CONSTANTS } from '../../constants/TIMEOUT_CONSTANTS';
+import { DriverHelper } from '../../utils/DriverHelper';
+import { error } from 'selenium-webdriver';
+
+suite(
+	`Create a workspace via launching a factory from the ${FACTORY_TEST_CONSTANTS.TS_SELENIUM_FACTORY_GIT_REPO_URL} repository`,
+	function (): void {
+		const browserTabsUtil: BrowserTabsUtil = e2eContainer.get(CLASSES.BrowserTabsUtil);
+		const workspaceHandlingTests: WorkspaceHandlingTests = e2eContainer.get(CLASSES.WorkspaceHandlingTests);
+		const projectAndFileTests: ProjectAndFileTests = e2eContainer.get(CLASSES.ProjectAndFileTests);
+		const loginTests: LoginTests = e2eContainer.get(CLASSES.LoginTests);
+		const pathToPluginRegistry: string = '/projects/devspaces/dependencies/che-plugin-registry/';
+		const internalRegistry: string = 'image-registry.openshift-image-registry.svc:5000';
+		const testWorkspaceUtil: ITestWorkspaceUtil = e2eContainer.get(TYPES.WorkspaceUtil);
+		const driverHelper: DriverHelper = e2eContainer.get(CLASSES.DriverHelper);
+		const viewItems: ViewItem[] | undefined = [];
+		const testRepoProjectName: string = 'devspaces';
+		const appVersion: string = BASE_TEST_CONSTANTS.TESTING_APPLICATION_VERSION?BASE_TEST_CONSTANTS.TESTING_APPLICATION_VERSION.replace(/(\.\d+)$/, ''):'next'
```

Comment:
Please, make sure regexp is not replacing minor version if `TESTING_APPLICATION_VERSION` doesn't have bug-fix version:
```
"3.9.0".replace(/(\.\d+)$/,'')
'3.9'
"3.9".replace(/(\.\d+)$/,'')
'3'
```